generators/docker-compose/index.js

```diff
@@ -0,0 +1,264 @@
+                    var config = fileData['generator-jhipster'];
+                    this.appConfigs.push(config);
+                }
+
+                done();
+            }.bind(this));
+        },
+
+        askForElk: function() {
+            if(this.abort) return;
+            var done = this.async();
+
+            var prompts = [{
+                type: 'confirm',
+                name: 'elk',
+                message: 'Do you want ELK to monitor your applications ?',
+                default: this.useElk && true
+            }];
+
+            this.prompt(prompts, function(props) {
+                this.useElk = props.elk;
+
+                done();
+            }.bind(this));
+        },
+
+        askForProfile: function() {
+            if(this.abort) return;
+            var done = this.async();
+
+            var choices = ['dev', 'prod'];
+
+            var prompts = [{
+                type: 'list',
+                name: 'profile',
+                message: 'Which profile do you want to use ?',
+                choices: choices,
+                default: this.profile || 'dev'
+            }];
+
+            this.prompt(prompts, function(props) {
+                this.profile = props.profile;
+
+                done();
+            }.bind(this));
+        }
+    },
+
+    configuring: {
+        checkImages: function() {
+            if(this.abort) return;
+
+            this.log('\nChecking Docker images in applications directories...');
+
+            for (var i = 0; i < this.appsFolders.length; i++) {
+                var imagePath = this.destinationPath(this.directoryPath + this.appsFolders[i] + '/target/docker/' + _.kebabCase(this.appConfigs[i].baseName) + '-0.0.1-SNAPSHOT.war');
+                if (!shelljs.test('-f', imagePath)) {
+                    this.log(chalk.red('\nDocker Image not found at ' + imagePath));
+                    this.log(chalk.red('Please run "mvn package docker:build" in ' + this.destinationPath(this.directoryPath + this.appsFolders[i]) + ' to generate Docker image'));
+                    this.abort = true;
+                }
+            }
+
+            if(!this.abort) this.log(chalk.green('Found Docker images, writing files...\n'));
+        },
+
+        generateJwtSecret: function() {
+            if(this.jwtSecretKey === undefined) {
+                this.jwtSecretKey = crypto.randomBytes(20).toString('hex');
+            }
+        },
+
+        setAppsFolderPaths: function() {
+            this.appsFolderPaths = [];
+            for (var i = 0; i < this.appsFolders.length; i++) {
+                var path = this.destinationPath(this.directoryPath + this.appsFolders[i]);
+                this.appsFolderPaths.push(path);
+            }
+        },
+
+        saveConfig: function() {
+            if(this.abort) return;
+            this.config.set('appsFolders', this.appsFolders);
+            this.config.set('appConfigs', this.appConfigs);
```

Comment:
do we have to save each apps config as it would be present on the path anyway and we can read from there always since we know `appsFolders`